Feature/ab#23 user profile page

BerendWouters opened the pull request:
Included:

- Account profile page
- Password reset in account profile page
- Overview of related care users

Joren-Thijs committed:
feat(user-profile-page) generated new components

Joren-Thijs committed:
feat(user-profile-page) finished base component
Added new navbar and router links

Joren-Thijs committed:
feat(user-profile-page) Added routerlinks and responsivenes

Joren-Thijs committed:
Adjusted media query

Joren-Thijs committed:
feat(user-profile-page) finished base component

AB#23

Joren-Thijs committed:
feat(user-profile-page) Added AccountInfo Service

AB#23

Joren-Thijs committed:
feat(user-profile-page) Removed AccountInfoService

AB#23

Joren-Thijs committed:
feat(user-profile-page) styled user profile page

AB#23

Joren-Thijs committed:
Merge branch 'develop' into feature/AB#23-user-profile-page

Joren-Thijs committed:
feat(user-profile-page) formatted code

AB#23

Joren-Thijs committed:
feat(user-profile-page) improved responsivenes
foraccount info page

Joren-Thijs committed:
Merge branch 'develop' into feature/AB#23-user-profile-page

Joren-Thijs committed:
feat(user-profile-page) added userInfoService

AB#23

BerendWouters committed:
update package.json version

BerendWouters committed:
chore adds GetUserId via PrincipalAccessor

AB#23

BerendWouters committed:
feat(user-profile) creates usercontroller

BerendWouters committed:
Merge branch 'feature/AB#23-user-profile-page' of https://github.com/Brixel/Singer into feature/AB#23-user-profile-page

# Conflicts:
#	Singer.API/ClientApp/package-lock.json
#	Singer.API/ClientApp/package.json

BerendWouters committed:
feat(user-profile) gets relevant data from api

AB#23

BerendWouters committed:
feat(reset-password) adds reset password functionality

AB#23

BerendWouters committed:
chore rename files to match page vs components

BerendWouters committed:
Merge branch 'develop' of https://github.com/Brixel/Singer into feature/AB#23-user-profile-page

# Conflicts:
#	Singer.API/ClientApp/package-lock.json
#	Singer.API/ClientApp/package.json

vyruz1986 commented on Singer.API/ClientApp/src/app/modules/core/components/nav-menu/nav-menu.component.html:
```diff
@@ -34,14 +26,21 @@
+
+         <button mat-menu-item *ngIf="isAuthenticated" [routerLink]="'/dashboard/account/account-info'">
+            <mat-icon>account_circle</mat-icon>
+            <span>Account</span>
```
Should be 'Profile'?

vyruz1986 replied:
'Profiel' I meant :)

BerendWouters committed:
fix use correct selectors

BerendWouters committed:
use Mijn profiel instead of Account

BerendWouters committed:
feat(profile) use "Profiel" instead of "Mijn profiel"

AB#23

vyruz1986 commented:
@BerendWouters visiting profile as admin user shows no information, is that expected? The only useful function for admin users I would expect there is to change your password, but that's not shown either. So I would suggest to either hide the profile menu item completely for admins (they can still reset their password via the admins list page), or show only personal information (hide careusers) and reset password link.

BerendWouters committed:
fix routing to account-info

vyruz1986 commented:
the `AccountInfo` components have been put under `SharedModule`, which isn't correct IMHO, since they're not reusable in other modules. They should be under `CoreModule`, same as NavMenu and About

vyruz1986 committed:
fix(UserProfile): Show text if LGUser has no careUsers lings

Closes AB#23

vyruz1986 committed:
style(UserProfile): Fixed some minor language-related issues

Better terminology for password reset.
Changed snackbar notification text for password reset into a more
descriptive text, and changed notification timeout to 10 seconds so user
has time to read it

Closes AB#23

vyruz1986 committed:
refactor(DetailedUserCard): Small improvement on showing buttons logic

Moved check to show buttons to parent div instead of on both child
buttons

vyruz1986 committed:
fix(UserProfile): Only show CareUsers for LGUsers

We now only show the CareUsers menu item in the user profile when the logged in user is a legal guardian.
Also noticed a bug when creating new LGUsers, the LG role was not attached to them.

vyruz1986 committed:
fix(Startup): Added check for all existing LGUsers

This check will make sure during startup, that all LGusers have the
LGUser role

Closes AB#23

vyruz1986 committed:
refactor: Cleaned up console.logs
